### PR TITLE
User

### PR DESCRIPTION
**Pull request by AlexMegas**
Created "user" branch.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -4,7 +4,8 @@ const mongoose = require('mongoose');
 const morgan = require('morgan');
 const bodyParser = require('body-parser');
 const expressValidator = require('express-validator');
-const route = require("./routes/book.js");
+const book = require("./routes/book.js");
```
Comment (IvanBurychka) [inline app.js]: name should be like bookRoute ...

File: app.js [hunk 2]
```diff
@@ -21,7 +22,9 @@ mongoose.connection.on('error', err => console.log(`MongoDB connection error ${e
 app.use(morgan('dev'));
 app.use(bodyParser.json());
 app.use(expressValidator());
-app.use("/api", route);
+app.use("/api", book);
```
Comment (IvanBurychka) [inline app.js]: route should be /api/books

File: app.js [hunk 3]
```diff
@@ -21,7 +22,9 @@ mongoose.connection.on('error', err => console.log(`MongoDB connection error ${e
...
+app.use("/user", user);
```
Comment (IvanBurychka) [inline app.js]: route should be /api/users

File: controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const user = require('../models/userSchema.js');
+
+exports.signup = (req, res) => {
+    console.log(req.body);
+    // async await
+    // to be continue...
```
Comment (IvanBurychka) [inline controllers/user.js]: try send response

File: models/userSchema.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require('mongoose');
+const crypto = require('crypto');
+const jwt = require('jsonwebtoken');
```
Comment (IvanBurychka) [inline models/userSchema.js]: for now we will use simple token without jwt

File: models/userSchema.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+const userSchema = new mongoose.Schema({
+    name: {
+        type: String,
+        required: true,
+        dropDups: true
```
Comment (IvanBurychka) [inline models/userSchema.js]: dropDups: true - why you use this?

File: models/userSchema.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    },
+    passw: {
+        type: String,
+        required: true
+    },
+    hash: String,
```
Comment (IvanBurychka) [inline models/userSchema.js]: what is 'hash'

File: app.js (hunk 1 above)
Comment (AlexMegas) [reply app.js]: Fixed: from "book" to "bookRoute"

File: app.js (hunk 2 above)
Comment (AlexMegas) [reply app.js]: Route for books changed to:  app.use("/api/books", bookRoute);

File: app.js (hunk 3 above)
Comment (AlexMegas) [reply app.js]: Fixed:   app.use("/api/users", userRoute);

File: controllers/user.js (hunk 4 above)
Comment (AlexMegas) [reply controllers/user.js]: Response sent successfully. As example:
{
    "name": "alex",
    "passw": "qwerty123"
}

File: models/userSchema.js (hunk 5 above)
Comment (AlexMegas) [reply models/userSchema.js]: Ok.

File: models/userSchema.js (hunk 6 above)
Comment (AlexMegas) [reply models/userSchema.js]: To clarify as unique. For removing Duplicates from MongoDB.

File: models/userSchema.js (hunk 7 above)
Comment (AlexMegas) [reply models/userSchema.js]: it is a unique string of fixed size.

**Review by AlexMegas — COMMENTED** [review]
reviewed and fixed.

**Comment by IBurychka** [comment]
{new: true}

book - books

onst user = new User(req.body);
        if (await User.findOne({ name: req.body.name }) === null) {

bCrypt.hash(req.body.passw, 10, (err, hash) => {

 if (result) {
                            //token ?

return res.status(200).send('User logged out :)');

check if user can delete book

/token ?
                            res.status(200).send('User logged in ;)');
how you know is user log in?

**Pull request merged**